### PR TITLE
Add verify

### PR DESCRIPTION
**Pull request by acured**
Please use `[API Name] Brief description` as title.

## API Name (Kind)
`Face`, `TextAnalytics`, `ComputerVision`...

## Purpose
Describe the intention of the changes being proposed.
What problem does it solve or functionality does it add?

## Other Information
Add any other helpful information that may be needed here.

### CODE REVIEW

File: Face/Verify/Program.cs [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+using System.Threading.Tasks;
+using Microsoft.Azure.CognitiveServices.Vision.Face;
+using Microsoft.Azure.CognitiveServices.Vision.Face.Models;
+
+namespace Verify
+{
+    public class Program
+    {
+        // Create a client. 
+        static string ApiKey = "ENTER YOUR KEY HERE";
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `public`

File: Face/Verify/Program.cs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+        static IFaceClient Client = new FaceClient(new ApiKeyServiceClientCredentials(ApiKey))
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `public`

File: Face/Verify/Program.cs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+        public static async Task Verify_FaceToFace()
+        {
+            List<Guid> targetFaceIds = new List<Guid>();
+            Guid sourceFaceId1 = new Guid();
+            Guid sourceFaceId2 = new Guid();
+
+            foreach (var imageFileName in TargetImageFileNames)
+            {
+                // Read image file.
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: // Read target image files.

File: Face/Verify/Program.cs [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+                    {
+                        Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                        return;
+                    }
+
+                    targetFaceIds.Add(detectedFaces[0].FaceId.Value);
+                }
+            }
+
+            // Read image file.
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: // Read source image file 1.

File: Face/Verify/Program.cs [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+                if (detectedFaces[0].FaceId == null)
+                {
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                    return;
+                }
+
+                sourceFaceId1 = detectedFaces[0].FaceId.Value;
+            }
+
+            // Read image file.
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: // Read source image file 2.

File: Face/Verify/Program.cs [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+            Verify_in_LargePersonGroup().Wait();
+
+            Console.WriteLine("\nPress ENTER to exit.");
+            Console.ReadLine();
+        }
+
+        public static async Task Verify_FaceToFace()
+        {
+            List<Guid> targetFaceIds = new List<Guid>();
+            Guid sourceFaceId1 = new Guid();
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `            Guid sourceFaceId1;`
`            Guid sourceFaceId2;`

File: Face/Verify/Program.cs [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+            Console.WriteLine(
+                verifyResult2.IsIdentical
+                    ? $"Faces from {SourceImageFileName2} & {TargetImageFileNames[0]} are of the same (Negative) person, similarity confidence: {verifyResult2.Confidence}."
+                    : $"Faces from {SourceImageFileName2} & {TargetImageFileNames[0]} are of different (Positive) persons, similarity confidence: {verifyResult2.Confidence}.");
+        }
+
+        public static async Task Verify_in_PersonGroup()
+        {
+            // Create a person group.
+            string GroupId = Guid.NewGuid().ToString();
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `personGroupId`

File: Face/Verify/Program.cs [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+            Console.WriteLine($"Create a person group ({GroupId}) {Environment.NewLine}");
+            await Client.PersonGroup.CreateAsync(GroupId, GroupId);
+
+            // Create a person group person.
+            Person p = new Person();
+            p.Name = "Dad";
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `Person p = new Person { Name = "Dad", UserData = "Person for sample" };`

File: Face/Verify/Program.cs [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+                if (detectedFaces[0].FaceId == null)
+                {
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                    return;
+                }
+
+                sourceFaceId2 = detectedFaces[0].FaceId.Value;
+            }
+
+
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: Leave only 1 newline.

File: Face/Verify/Program.cs [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for Identify purpose.");
+                    return;
+                }
+
+                // Add detected faceId to faceIds.
+                faceIds.Add(detectedFaces[0].FaceId.Value);
+            }
+
+            // Verification example for faces of the same person.
+            VerifyResult VerifyResults = await Client.Face.VerifyFaceToPersonAsync(faceIds[0], p.PersonId, GroupId);
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `verifyResult`

File: Face/Verify/Program.cs [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+
+            // Delete the person group.
+            Console.WriteLine($"{Environment.NewLine}Delete the person group ({GroupId})");
+            await Client.PersonGroup.DeleteAsync(GroupId);
+        }
+
+        public static async Task Verify_in_LargePersonGroup()
+        {
+            // Create a large person group.
+            string GroupId = Guid.NewGuid().ToString();
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `largePersonGroupId`

File: Face/Verify/Program.cs [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+            Console.WriteLine($"Create a large person group ({GroupId}) {Environment.NewLine}");
+            await Client.LargePersonGroup.CreateAsync(GroupId, GroupId);
+
+            // Create a large person group person.
+            Person p = new Person();
+            p.Name = "Dad";
+            p.UserData = "Person for sample";
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `Person p = new Person { Name = "Dad", UserData = "Person for sample" };`

File: Face/Verify/Program.cs [hunk 13]
```diff
@@ -0,0 +1,289 @@
...
+            Console.WriteLine($"Create a person group person '{p.Name}' {Environment.NewLine}");
+            p.PersonId = (await Client.PersonGroupPerson.CreateAsync(GroupId, p.Name)).PersonId;
+
+            foreach (var targetImageFileName in TargetImageFileNames)
+            {
+                // Read image file. 
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Add face to the person group. 
+                    Console.WriteLine($"Add faces to the person group from image `{targetImageFileName}`.");
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `Console.WriteLine($"Add face to the large person group person from image `{targetImageFileName}`.");`

Same for other places.

File: Face/Verify/Program.cs [hunk 14]
```diff
@@ -0,0 +1,289 @@
...
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for Identify purpose.");
+                    return;
+                }
+
+                // Add detected faceId to faceIds.
+                faceIds.Add(detectedFaces[0].FaceId.Value);
+            }
+
+            // Verification example for faces of the same person.
+            VerifyResult VerifyResults = await Client.Face.VerifyFaceToPersonAsync(faceIds[0], p.PersonId, null, GroupId);
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: `verifyResults`

File: Face/Verify/Program.cs [hunk 15]
```diff
@@ -0,0 +1,289 @@
...
+            while (true)
+            {
+                await Task.Delay(1000);
+                var status = await Client.PersonGroup.GetTrainingStatusAsync(GroupId);
+                Console.WriteLine($"Response: Success. Group \"{GroupId}' training process is {status.Status}");
+                if (status.Status != TrainingStatusType.Running)
+                {
+                    break;
+                }
+            }
```
Comment (lebronJ) [inline Face/Verify/Program.cs]: No need to train before verification. Both for PersonGroup and LargePersonGroup.

File: Face/Verify/Program.cs (hunk 1 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 2 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 3 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 4 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 5 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 6 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 7 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 8 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 9 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 10 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 11 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 12 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 13 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 14 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

File: Face/Verify/Program.cs (hunk 15 above)
Comment (acured) [reply Face/Verify/Program.cs]: done

**Pull request merged**